### PR TITLE
feat: root installation

### PR DESCRIPTION
**Pull request by CnC-Robert**

### CODE REVIEW

File: app/src/main/java/app/revanced/manager/domain/installer/RootInstaller.kt [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+    companion object {
+        const val modulesPath = "/data/adb/modules"
+
+        /*val isDeviceRooted =
+            try {
+                Runtime.getRuntime().exec("su --version")
+                true
+            } catch (_: IOException) {
+                false
+            }*/
```
Comment (Axelen123) [inline app/src/main/java/app/revanced/manager/domain/installer/RootInstaller.kt:128]: Why is this left here?

File: app/src/main/java/app/revanced/manager/ui/component/SegmentedButton.kt [hunk 2]
```diff
@@ -28,41 +30,54 @@ fun RowScope.SegmentedButton(
...
                 )
+                .weight(1f)
+                .padding(vertical = 20.dp)
+        ) {
+            when (icon) {
+                is ImageVector -> {
+                    Icon(
+                        imageVector = icon,
+                        contentDescription = iconDescription,
+                        //tint = MaterialTheme.colorScheme.primary,
```
Comment (Axelen123) [inline app/src/main/java/app/revanced/manager/ui/component/SegmentedButton.kt:61]: Why was this changed? Why is the comment left here?

File: app/src/main/java/app/revanced/manager/ui/screen/InstallerScreen.kt [hunk 3]
```diff
@@ -121,6 +130,50 @@ fun InstallerScreen(
...
+        confirmButton = {
+            Button(onClick = {
+                onConfirm(selectedInstallType)
+                onDismiss()
+            }
+            ) {
+                Text("Install")
+            }
+        },
+        title = { Text("Select installation type") },
```
Comment (Axelen123) [inline app/src/main/java/app/revanced/manager/ui/screen/InstallerScreen.kt:156]: Make the text translatable

File: app/src/main/java/app/revanced/manager/ui/viewmodel/AppSelectorViewModel.kt [hunk 4]
```diff
@@ -11,22 +11,19 @@ import java.nio.file.Files
...
     private val app: Application,
-    pm: PM
+    private val pm: PM
 ) : ViewModel() {
-    private val packageManager = app.packageManager
-
     val appList = pm.appList
 
-    fun loadLabel(app: PackageInfo?) = (app?.applicationInfo?.loadLabel(packageManager) ?: "Not installed").toString()
+    fun loadLabel(app: PackageInfo?) = with(pm) { app?.label() ?: "Not installed" }
```
Comment (Axelen123) [inline app/src/main/java/app/revanced/manager/ui/viewmodel/AppSelectorViewModel.kt:18]: The fallback message should be translatable

File: app/src/main/java/app/revanced/manager/ui/viewmodel/InstallerViewModel.kt [hunk 5]
```diff
@@ -192,20 +213,59 @@ class InstallerViewModel(input: Destination.Installer) : ViewModel(), KoinCompon
...
     }
 
-    fun installOrOpen() = viewModelScope.launch {
-        installedPackageName?.let {
+    fun install(installType: InstallType) = viewModelScope.launch {
+        /*installedPackageName?.let {
             pm.launch(it)
             return@launch
-        }
+        }*/
```
Comment (Axelen123) [inline app/src/main/java/app/revanced/manager/ui/viewmodel/InstallerViewModel.kt:220]: Remove the comment

**Review by Axelen123 — CHANGES_REQUESTED** [review]

File: app/src/main/java/app/revanced/manager/domain/installer/RootInstaller.kt (hunk 1 above)
Comment (CnC-Robert) [reply app/src/main/java/app/revanced/manager/domain/installer/RootInstaller.kt:128]: I forgot to remove it

File: app/src/main/java/app/revanced/manager/ui/component/SegmentedButton.kt (hunk 2 above)
Comment (CnC-Robert) [reply app/src/main/java/app/revanced/manager/ui/component/SegmentedButton.kt:61]: The content color should be gray when it is disabled. Im providing the color with CompositionLocalProvider now

Comment (Axelen123) [reply app/src/main/java/app/revanced/manager/ui/component/SegmentedButton.kt:61]: Fair enough. The comment should be removed though

File: app/src/main/java/app/revanced/manager/domain/installer/RootInstaller.kt [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+
+                                while (inputStream.read(buffer).also { bytesRead = it } != -1) {
+                                    val content = String(buffer, 0, bytesRead)
+                                        .replace("__PKG_NAME__", packageName)
+                                        .replace("__VERSION__", version)
+                                        .replace("__LABEL__", label)
+                                        .toByteArray()
+
+                                    outputStream.write(content)
+                                }
```
Comment (Axelen123) [inline app/src/main/java/app/revanced/manager/domain/installer/RootInstaller.kt:92]: This will cause problems in the future if the file is changed in such a way that `__PKG_NAME__` ends up at a 1024-byte boundary, because it will get cut off and not get replaced by the expected value.

Can be solved by reading the entire file instead of streaming (or by using something thats made for streams). Memory needs to be traded for correctness in this situation

**Review by Axelen123 — CHANGES_REQUESTED** [review]

**Review by Axelen123 — APPROVED** [review]
